.github/workflows/ci.yml

```diff
@@ -7,13 +7,18 @@ on:
     branches:
       - master
 jobs:
-  ci-test:
-    runs-on: ubuntu-latest
+  test:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: '12'
-      - run: npm i -g npm
+          cache: npm
```

Comment:
This may not work as-is as we use `npm ci` (wipes the folder); Our specific monorepo setup with `lerna bootstrap --hoist` may also cause some issues (e.g. [this](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies) probably has to be specified)

Let's move this to another PR where the performance impact is compared (and any downsides of loosing `npm ci`)

---

Lgtm otherwise!

I'll disable the appveyor checks once this is merged

Reply:
Hi @ang-zeyu, yup sure I can remove the caching. I just found out what `npm ci` is about 😂  ...it does feel like cache may not be utilized without proper configuration. By the way, is it ok to keep the update of `setup-node` from v1 to v2? I think there's no significant difference that impacts our use case but might be good to use an updated version.

Reply:
👍 

Ideally do try to keep unrelated changes in other commits / prs for collaboration clarity (e.g. someone scanning through the commit history of titles may not be aware of a version change here), I'm *personally* ok here since this is a small devops change.

Reply:
Yup duly noted!